src/test/scala/cognite/spark/v1/FileContentRelationTest.scala

```diff
@@ -0,0 +1,160 @@
+      _ <- existingFile match {
+        case Right(value) if !value.uploaded => writeClient.files.deleteByExternalIds(Seq(externalId))
+        case _ => IO.pure(())
+      }
+
+      file <- existingFile match {
+        case Right(value) if value.uploaded => IO.pure(value)
+        case _ =>
+          writeClient.files.create(Seq(toCreate)).map(_.headOption.getOrElse(throw new IllegalStateException("could not upload file")))
+      }
+
+      _ <- {
+        if (!file.uploaded) {
+          val backend = Resource.make(IO(HttpURLConnectionBackend()))(b => IO(b.close()))
+          val request = basicRequest.put(uri"${file.uploadUrl.getOrElse(throw new MalformedURLException("bad url"))}")
+            .body(generateNdjsonData)
+          backend.use { backend =>
+            IO {
+              val response = request.send(backend)
+              if (response.code.isSuccess) {
+                println(s"NDJSON content uploaded successfully to ${file.uploadUrl}")
+              } else {
+                throw new Exception(s"Failed to upload content: ${response.statusText}")
+              }
+            }
+          }
+        } else {
+          IO.pure(())
+        }
+      }
+    } yield ()
+  }
+
+  "file content transformations" should "read from a ndjson file" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("externalId", fileExternalId)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    val result = spark.sqlContext.sql(s"select * from filecontent").collect()
+    result.map(_.toSeq.toList) should contain theSameElementsAs
+      Array(
+        List(30, "Alice", null),
+        List(25, "Bob", null),
+        List(35, "Charlie", null),
+        List(35, "Charlie2", "test")
+      )
+  }
+
+  it should "fail with a sensible error if the mimetype is wrong" in {
+    val exception = sparkIntercept {
+      val sourceDf: DataFrame = dataFrameReaderUsingOidc
+        .useOIDCWrite
+        .option("type", "filecontent")
+        .option("externalId", fileWithWrongMimeTypeExternalId)
+        .load()
+      sourceDf.createOrReplaceTempView("fileContent")
+      spark.sqlContext.sql(s"select * from filecontent").collect()
+    }
+    assert(exception.getMessage.contains("Wrong mimetype. Expects application/jsonlines"))
+  }
+
+  it should "infer the schema" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalId)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
+        StructField("name", StringType, nullable = true),
+        StructField("test", StringType, nullable = true),
+        StructField("age", LongType, nullable = true)
```

Comment:
maybe also be good to add tests with
- nested object types
- conflicting column types between rows